docker/Dockerfile.base

```diff
@@ -0,0 +1,277 @@
+
+RUN : \
+    && set -eux \
+	&& USER_MANUAL_INSTALLED="$(dnf history userinstalled | tail -n +2)" \
+	&& dnf upgrade -y \
+	&& dnf install -y wget findutils \
+	&& wget -O get-pip.py "$PYTHON_GET_PIP_URL" \
+	&& echo "$PYTHON_GET_PIP_SHA256 *get-pip.py" | sha256sum -c - \
+	&& dnf list installed | tail -n +2 | cut -d' ' -f1 | xargs -r dnf mark remove > /dev/null \
+	&& [ -z "$USER_MANUAL_INSTALLED" ] || dnf mark install $USER_MANUAL_INSTALLED > /dev/null \
+	&& dnf autoremove -y \
+	&& python get-pip.py \
+		--disable-pip-version-check \
+		--no-cache-dir \
+		--no-compile \
+		"pip==$PYTHON_PIP_VERSION" \
+		"setuptools==$PYTHON_SETUPTOOLS_VERSION" \
+	&& rm -f get-pip.py \
+	\
+    && dnf clean all \
+    && rm -rf /var/cache/yum \
+    && rm -rf /var/cache/dnf \
+	&& pip --version
+
+# INSTALLING SOUFFLE
+# The instructions to build Souffle in Oracle Linux is referenced from:
+#       https://github.com/souffle-lang/souffle/blob/master/.github/images/oraclelinux-8/Dockerfile
+RUN : \
+    && set -eux \
+    && mkdir /souffle-src \
+    && cd /souffle-src \
+    && USER_MANUAL_INSTALLED="$(dnf history userinstalled | tail -n +2)" \
+    && dnf upgrade -y \
+    && dnf --enablerepo=ol9_codeready_builder install -y \
+        autoconf \
+        automake \
+        bash-completion \
+        bison \
+        clang \
+        cmake \
+        doxygen \
+        flex \
+        gcc-c++ \
+        git \
+        libffi-devel \
+        libtool \
+        make \
+        # The package mcpp could not be found anywhere in available oracle linux 9 repos.
+        # However, souffle could be built without it.
+        # mcpp
+        ncurses-devel \
+        pkg-config \
+        python39 \
+        rpm-build \
+        sqlite-devel \
+        zlib-devel \
+    && git clone --depth=1 https://github.com/souffle-lang/souffle/ -b 2.3 \
+    && cd souffle \
+    && cmake -S . -B build -DSOUFFLE_DOMAIN_64BIT=ON -DCMAKE_INSTALL_PREFIX="/usr/local" \
+    && nproc="$(nproc)" \
+    && cmake --build build --target install -j "$nproc" \
+    && echo $USER_MANUAL_INSTALLED \
+    && dnf list installed | tail -n +2 | cut -d' ' -f1 | xargs -r dnf mark remove > /dev/null \
+	&& [ -z "$USER_MANUAL_INSTALLED" ] || dnf mark install $USER_MANUAL_INSTALLED > /dev/null \
+    # We exclude systemd-udev because one of the build dependencies for Souffle pull this protected package.
+    # This command make sure that it won't break if systemd-udev is not pulled in the future.
+    && (dnf list installed systemd-udev && dnf mark install systemd-udev || true) \
+	&& dnf autoremove -y \
+    && cd / \
+    && rm -rf /souffle-src \
+    && dnf clean all \
+    && rm -rf /var/cache/yum \
+    && rm -rf /var/cache/dnf \
+    && souffle --version
+
+# INSTALLING MACARON RUNTIME DEPENDENCIES AND OTHER UTILITIES PACKAGES.
+#   - java-17-openjdk-devel
+#   - git
+#   - findutils
+ENV JAVA_HOME="/usr/lib/jvm/java-17-openjdk"
```

Comment:
This is not correct as we are now using the Oracle JDK 17.

Reply:
Fixed in https://github.com/oracle-samples/macaron/pull/76/commits/b12c92fbb627a4d3d995e124a6e5846439aa6fcd.